### DISCUSSION TITLE
OTEL Instrumentation

### DISCUSSION THREAD

**Discussion by ghost** [kubernetes-client/csharp#898] [Ideas]
Hi,

Are there any plans to implement an OTEL instrumentation library for the kl8s client?

Thanks!

**Comment by tg123** [comment]
could you please elaborate it?
is prometheus good for you?

**Comment by brendandburns** [comment]
OTEL is open telemetry which would do request tracing, vs prometheus which does monitoring.

**Comment by brendandburns** [comment]
Oh, and we don't have current plans, but we'd be happy to take PRs I think.

**Comment by brendandburns** [comment]
Actually, since this client just uses HttpClient under the hood, I think that this:

https://opentelemetry.io/docs/instrumentation/net/automatic/

should just work?

**Comment by EraYaN** [comment]
The biggest problem right now is that this library has a direct dependency on prometheus-net. That should probably be in a separate assembly or use any of the native stuff for instrumentation (System.Diagnostics.Metrics I believe).
